RevenueCatUI/Templates/Components/Text/TextComponentView.swift

```diff
@@ -20,22 +20,39 @@ import SwiftUI
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 struct TextComponentView: View {
 
+    @Environment(\.componentViewState)
+    private var componentViewState
+
+    @Environment(\.screenCondition)
+    private var screenCondition
+
     private let viewModel: TextComponentViewModel
 
     internal init(viewModel: TextComponentViewModel) {
         self.viewModel = viewModel
     }
 
     var body: some View {
-        Text(viewModel.text)
-            .font(viewModel.textStyle)
-            .fontWeight(viewModel.fontWeight)
-            .fixedSize(horizontal: false, vertical: true)
-            .multilineTextAlignment(viewModel.horizontalAlignment)
-            .foregroundStyle(viewModel.color)
-            .padding(viewModel.padding)
-            .background(viewModel.backgroundColor)
-            .padding(viewModel.margin)
+        viewModel.styles(
+            state: self.componentViewState,
+            condition: self.screenCondition
+        ) { style in
```

Comment:
This is a new `@ViewBuilder` on the view model that creates a `TextComponentStyle` object to use for styling the component 👇

Reply:
I love it!